signac/core/collection_json.py

```diff
@@ -57,7 +121,7 @@ def _sync(self):
                 file.write(blob)
 
 
-class JSONDict(JSONCollection, SyncedAttrDict):
+class JSONDict(SyncedAttrDict, JSONCollection):
```

Comment:
Is there a reason this inheritance order needed to change? The design of the classes was intended to avoid any collisions that might require nonintuitive method resolution order.

Reply:
Having read the whole PR, I assume the reason has to do with the validators associated with the `SyncedAttrDict` class. I think this change can be reverted, see my comment below.